ocs_ci/helpers/dr_helpers.py

```diff
@@ -549,3 +561,127 @@ def get_all_drpolicy():
+    assert drcluster_obj.patch(
+        params=unfence_params, format_type="merge"
+    ), f"Failed to patch {constants.DRCLUSTER}: {drcluster_name}"
+    logger.info(f"Successfully unfenced {constants.DRCLUSTER}: {drcluster_name}")
+    config.switch_ctx(restore_index)
+
+
+def get_fence_state(drcluster_name):
+    """
+    Returns the clusterfence state of given drcluster
+
+    Args:
+        drcluster_name (str): Name of the DRcluster
+
+    Returns:
+        state (str): If drcluster are fenced: Fenced or Unfenced,
+                     else None if not defined
+
+    """
+    restore_index = config.cur_index
+    config.switch_acm_ctx()
+    drcluster_obj = ocp.OCP(resource_name=drcluster_name, kind=constants.DRCLUSTER)
+    state = drcluster_obj.get().get("spec").get("clusterFence")
+    config.switch_ctx(restore_index)
+    return state
+
+
+def validate_data_integrity(namespace, path="/mnt/test/hashfile", timeout=600):
+    """
+    Verifies the md5sum values of files are OK
+
+    Args:
+        namespace (str): Namespace where the workload running
+        path (str): Path of the hashfile saved of each files
+        timeout (int): Time taken in seconds to run command inside pod
+
+    Raises: If there is a mismatch in md5sum value or None
+
+    """
+    all_pods = get_all_pods(namespace=namespace)
+    for pod_obj in all_pods:
+        logger.info("Verify the md5sum values are OK")
+        cmd = f"md5sum -c {path}"
+        try:
+            pod_obj.exec_cmd_on_pod(command=cmd, out_yaml_format=False, timeout=timeout)
+            logger.info(f"Pod {pod_obj.name}: All files checksums value matches")
+        except CommandFailed as ex:
+            if "computed checksums did NOT match" in str(ex):
+                logger.error(
+                    f"Pod {pod_obj.name}: One or more files or datas are modified"
+                )
+            raise ex
+
+
+def gracefully_reboot_nodes(namespace, drcluster_name):
+    """
+    Gracefully reboot OpenShift Container Platform
+    nodes which was fenced before
+
+    Args:
+        namespace (str): Name of the namespace
+        drcluster_name (str): Name of the drcluster which need to be reboot
+
+    """
+
+    primary_cluster_name = get_current_primary_cluster_name(namespace=namespace)
+    if primary_cluster_name == drcluster_name:
+        set_current_primary_cluster_context(namespace)
+    else:
+        set_current_secondary_cluster_context(namespace)
+    node_objs = get_node_objs()
+    factory = platform_nodes.PlatformNodesFactory()
+    nodes = factory.get_nodes_platform()
+    waiting_time = 30
+    for node in node_objs:
+        node_name = node.name
+        unschedule_nodes([node_name])
+        drain_nodes([node_name])
+        nodes.restart_nodes([node], wait=False)
+        logger.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+        schedule_nodes([node_name])
+    wait_for_nodes_status(status=constants.NODE_READY, timeout=180)
```

Comment:
Would it be possible to reuse `gracefully_reboot_nodes()` function from the nodes module (which you added in this PR)?

Reply:
done